src/mesh_analysis/tracers/waku_tracer.py

```diff
@@ -15,21 +20,108 @@ def __init__(self):
         # - Different patterns (received, sent, dropped)
         # - Once one pattern search is completed, stop search for it in the logs (ie: Announce Address)
         super().__init__()
-        self._patterns = [r'my_peer_id=([\w*]+).*?msg_hash=(0x[\da-f]+).*?receivedTime=(\d+)']
+        self._patterns = []
+        self._tracings = []
 
-    def trace(self, parsed_logs: List) -> pd.DataFrame:
-        df = self._trace_message_in_logs(parsed_logs)
+    def with_received_pattern(self):
```

Comment:
Maybe you want to consider returning `self` for these `with_` to chain these functions with the _Fluent Interface_ pattern. I don't know what's the intended usage for them, and it might not be useful for this case, but I thought I'd mention it. Obviously, up to you :)

Reply:
Yeah, I like that, will change it.

Reply:
_Builder_ is a pattern that commonly uses this one, yes; but that doesn't mean you can't leverage it for other use cases as well. Of course, if you rather not use it that's totally fine, just figured I'd mention it just in case :)